packages/blade/src/components/FileUpload/FileUpload.web.tsx

```diff
@@ -0,0 +1,323 @@
+import { useState, useCallback, useMemo, forwardRef, memo } from 'react';
+import type { FileUploadProps, BladeFile, BladeFileList } from './types';
+import { StyledFileUploadWrapper } from './StyledFileUploadWrapper';
+import { fileUploadColorTokens, getFileUploadInputHoverTokens } from './fileUploadTokens';
+import { FileUploadItem } from './FileUploadItem';
+import BaseBox from '~components/Box/BaseBox';
+import { Box } from '~components/Box';
+import { SelectorLabel } from '~components/Form/Selector/SelectorLabel';
+import { SelectorInput } from '~components/Form/Selector/SelectorInput';
+import { screenReaderStyles } from '~components/VisuallyHidden';
+import { FormHint, FormLabel } from '~components/Form';
+import { useFormId } from '~components/Form/useFormId';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { getStyledProps } from '~components/Box/styledProps';
+import { makeSize, useTheme } from '~utils';
+import { Text } from '~components/Typography';
+import type { BladeElementRef } from '~utils/types';
+import { getHintType } from '~components/Input/BaseInput/BaseInput';
+import { makeAccessible } from '~utils/makeAccessible';
+
+const MemoizedFileUploadItem = memo(FileUploadItem);
```

Comment:
Why do we need to memoize it here? 

Can we not just export a memoized FileInput if this is needed?

Reply:
yes, done.